### ISSUE TITLE
Token expires instantly when there are two access points

### ISSUE THREAD

**Issue by khanzadyan** [elasticio/sugarcrm-component#19]
There is an issue with access_token expiration which can affect almost every integrator using this component. The error is reported during the integration execution:
```
Error: Authentication error: invalid_grant: The access token provided is invalid.
    at Service.makeRequest (/app/lib/sugarcrm.js:191:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)
```
Here is a standard scenario:

1. OAuth2 app is created on Sugar CRM admin panel.
2. Credentials are created on elastic.io using the details from step1
3. A simple integration flow like WebHook to SugarCRM is created and activated.
4. We send a payload to WebHook, the integration flow starts working and successfully transferring the payload to SugarCRM.
5. We go back to Sugar CRM to check if the data is there. The data is there. 
6. We want to try it again to make sure that everything works OK but bummer, we get the error shown above while the Sugar CRM component tried to access the Sugar CRM instance.

**It is as if the token got expired in minutes!**

I found the reason for this: the system detects that you are logged-in from your browser (you were checking the data in Sugar CRM) and in addition somebody is trying to access using OAuth2 from somewhere else (our platform). Some additional information can be found here:
* https://community.sugarcrm.com/thread/31763-why-does-the-api-return-invalidgrant-before-my-access-token-has-expired
* http://support.sugarcrm.com/Knowledge_Base/Troubleshooting/Troubleshooting_Sugar_Logging_Out_Unexpectedly/#Resolution

Some temporary solution could be to tell explicitly not to use the same credentials (the admin) during the flow execution, but this is not going to work since people who do the integrations are busy preparing integrations for some other client and creating different credentials might not be an option.

**Comment by Nazar910** [comment]
http://support.sugarcrm.com/Documentation/Sugar_Developer/Sugar_Developer_Guide_7.8/Integration/Web_Services/v10/.
The trouble is with parameter `platform`. From link `Normally, when logging into Sugar, users login with a platform type of "base". We are using "custom" to avoid any potential login conflicts.`

**Comment by drobiazko** [comment]
@Nazar910 I guess `custom` is just an example. You could use any string, right? Maybe `elasticio`?

**Comment by Nazar910** [comment]
@drobiazko yep, you're right

**Comment by jhorbulyk** [comment]
Solved by #20
